entity_api/src/coaching_session.rs

```diff
@@ -0,0 +1,121 @@
+use super::error::{EntityApiErrorCode, Error};
+use crate::{naive_date_parse_str, uuid_parse_str};
+use entity::coaching_sessions::{self, Entity, Model};
+use sea_orm::{entity::prelude::*, DatabaseConnection};
+use std::collections::HashMap;
+
+pub async fn find_by(
+    db: &DatabaseConnection,
+    params: HashMap<String, String>,
+) -> Result<Vec<Model>, Error> {
+    let mut query = Entity::find();
+
+    for (key, value) in params {
+        match key.as_str() {
+            "coaching_relationship_id" => {
```

Comment:
@calebbourg I'm curious if there's a way of getting the key name as a string to compare to here, i.e. "coaching_relationship_id", "from_date", etc by adding these as constant string literals in entity/src/coaching_sessions.rs and referring to them here by constant name. If this is tricky and thus takes a lot of time, don't worry about it then. I was just thinking if it's straightforward we might as well make this change.

Reply:
That's a good idea. I'll give it a try

Reply:
So for each `entity` the `DeriveEntityModel` adds a `Column` enum that has a variant for each of the table columns.  An example would be `coaching_relationship::Column::CoachingRelationshipId`. This, in my mind would be the best candidate to convert into strings or `&str` in order to compare in these `entity_api` functions. Unfortunately, there isn't a good built in trait for converting enums to `&str` or `&'static str`. There are a few crates that provide implementations like [strum](https://github.com/Peternator7/strum) though I think for now I'd like to hold off on adding the dependency and see if this pattern that we're using here stabilizes in our project.

Reply:
Sounds good Caleb, thanks for looking into it it a little bit.